packages/page-addresses/src/modals/Create.tsx

```diff
@@ -45,7 +45,7 @@ function Create ({ onClose, onStatusChange }: Props): React.ReactElement<Props>
 
         address = keyring.encodeAddress(publicKey);
         isAddressValid = keyring.isAvailable(address);
-        isPublicKey = publicKey.length === 32;
+        isPublicKey = isEthereum? publicKey.length === 20:publicKey.length === 32;
```

Comment:
I believe the Ethereum public key is 32 bytes (64 in reality but only 32 are useful/needed). It then gets converted to 20 bytes to retrieve the address (AccountId)

Reply:
For Moonbeam (ethereum type) I believe we actually have the full 65 bytes uncompressed.